frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/upload-pvc-form/upload-pvc-form.tsx

```diff
@@ -0,0 +1,382 @@
+import cx from 'classnames';
+import { Helmet } from 'react-helmet';
+import { FileUpload, ActionGroup, Alert, Button } from '@patternfly/react-core';
+import { isCephProvisioner, isObjectSC } from '@console/shared/src/utils';
+import { K8sResourceKind, apiVersionForModel } from '@console/internal/module/k8s';
+import {
+  ButtonBar,
+  RequestSizeInput,
+  history,
+  resourcePath,
+} from '@console/internal/components/utils';
+import { StorageClassDropdown } from '@console/internal/components/utils/storage-class-dropdown';
+import { RadioInput } from '@console/internal/components/radio';
+import { StorageClass } from '@console/internal/components/storage-class-form';
+import { DataVolumeModel } from '../../../models';
+import { createUploadPVC } from '../../../k8s/requests/cdi-upload/cdi-upload-requests';
+import { CDIUploadContext } from '../cdi-upload-provider';
+import { UploadPVCFormStatus } from './upload-pvc-form-status';
+import { PersistentVolumeClaimModel } from '@console/internal/models';
+import { getName, getNamespace } from '@console/shared';
+import { V1alpha1DataVolume } from '../../../types/vm/disk/V1alpha1DataVolume';
+import './upload-pvc-form.scss';
+
+const cephRBDProvisionerSuffix = 'rbd.csi.ceph.com';
+
+// See https://kubernetes.io/docs/concepts/storage/persistent-volumes/#types-of-persistent-volumes for more details
+const provisionerAccessModeMapping = {
+  'kubernetes.io/no-provisioner': ['ReadWriteOnce'],
+  'kubernetes.io/aws-ebs': ['ReadWriteOnce'],
+  'kubernetes.io/gce-pd': ['ReadWriteOnce', 'ReadOnlyMany'],
+  'kubernetes.io/glusterfs': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
+  'kubernetes.io/cinder': ['ReadWriteOnce'],
+  'kubernetes.io/azure-file': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
+  'kubernetes.io/azure-disk': ['ReadWriteOnce'],
+  'kubernetes.io/quobyte': ['ReadWriteOnce', 'ReadWriteMany', 'ReadOnlyMany'],
+  'kubernetes.io/rbd': ['ReadWriteOnce', 'ReadOnlyMany'],
+  'kubernetes.io/vsphere-volume': ['ReadWriteOnce', 'ReadWriteMany'],
+  'kubernetes.io/portworx-volume': ['ReadWriteOnce', 'ReadWriteMany'],
+  'kubernetes.io/scaleio': ['ReadWriteOnce', 'ReadOnlyMany'],
+  'kubernetes.io/storageos': ['ReadWriteOnce'],
+};
+
+export const UploadPVCForm: React.FC<UploadPVCFormProps> = (props) => {
+  const [accessModeHelp, setAccessModeHelp] = React.useState('Permissions to the mounted drive.');
+  const [allowedAccessModes, setAllowedAccessModes] = React.useState([
+    'ReadWriteOnce',
+    'ReadWriteMany',
+    'ReadOnlyMany',
+  ]);
+
+  const [storageClass, setStorageClass] = React.useState('');
+  const [pvcName, setPvcName] = React.useState('');
+  const [accessMode, setAccessMode] = React.useState('ReadWriteOnce');
+  const [requestSizeValue, setRequestSizeValue] = React.useState('');
+  const [requestSizeUnit, setRequestSizeUnit] = React.useState('Gi');
+  const [storageProvisioner, setStorageProvisioner] = React.useState('');
+  const accessModeRadios = [
```

Comment:
we shouldn't duplicate this component, but make the original one (`create-pvc.tsx`) extensible or components shared. This would become a pain for maintainability.

Reply:
i created a `shared.ts` file with shared consts and methods between the forms